### ISSUE TITLE
Hovered text drops too far in contrast to be easily legible.

### ISSUE THREAD

**Issue by kevinreiss** [pulibrary/allsearch_frontend#322]
### Steps to recreate
Visit any search result page, i.e. https://allsearch.princeton.edu/?q=dogs. 

### User impact
Low vision users are challenged to use the site. 

### Severity
Medium per John Jameson. 

### WCAG Conformance level (A, AA, AAA)
Visual distinctions used to identify elements and states (e.g., a link's underline, its hover color and focus outline) each have a contrast ratio of at least 3:1 against their background and alternate states. (1.4.11 AA)

### Example
<img width="602" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOVdZQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--156f6fc42c15e520cb3d1e18a54b9b0d9a97b20c/Screenshot%202024-06-27%20at%208.35.20%E2%80%AFAM.png" alt="Screenshot 2024-06-27 at 8.35.20 AM.png" />

### Recommended fix
John's note: Orange is OK as a graphical element, so in theory you could just change the underline and icon color to orange and keep the text black. 

### Acceptance criteria
- [ ] I can successfully use this feature with Voice Over in Safari.
- [ ] I have checked this feature using one of the following browser extensions and confirmed that it does not add any level A or level AA issues:
    * axe dev tools
    * Lighthouse
    * Wave

**Comment by maxkadel** [comment]
We started looking at this as part of the work of #344
